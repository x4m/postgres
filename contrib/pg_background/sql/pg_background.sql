CREATE EXTENSION pg_background;

--run 6 workers which wait .0, .1, .2, .3, .4, .5 seconds respectively
CREATE TABLE input as SELECT x FROM generate_series(0,.5,0.1) x ORDER BY x DESC;
CREATE TABLE output(place int,value float);
--sequence for indication of who's finished on which place
CREATE sequence s start 1;
CREATE TABLE handles as SELECT pg_background_launch(format('select pg_sleep(%s); insert into output values (nextval(''s''),%s);',x,x)) h FROM input;
--wait until everyone finishes
SELECT (SELECT * FROM pg_background_result(h) as (x text) limit 1) FROM handles;
--output results
SELECT * FROM output ORDER BY place;
