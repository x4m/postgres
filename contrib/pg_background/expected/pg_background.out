CREATE EXTENSION pg_background;
--run 6 workers which wait .0, .1, .2, .3, .4, .5 seconds respectively
CREATE TABLE input as SELECT x FROM generate_series(0,.5,0.1) x ORDER BY x DESC;
CREATE TABLE output(place int,value float);
--sequence for indication of who's finished on which place
CREATE sequence s start 1;
CREATE TABLE handles as SELECT pg_background_launch(format('select pg_sleep(%s); insert into output values (nextval(''s''),%s);',x,x)) h FROM input;
--wait until everyone finishes
SELECT (SELECT * FROM pg_background_result(h) as (x text) limit 1) FROM handles;
    x     
----------
 SELECT 1
 SELECT 1
 SELECT 1
 SELECT 1
 SELECT 1
 SELECT 1
(6 rows)

--output results
SELECT * FROM output ORDER BY place;
 place | value 
-------+-------
     1 |     0
     2 |   0.1
     3 |   0.2
     4 |   0.3
     5 |   0.4
     6 |   0.5
(6 rows)

